upstairs/src/lib.rs

```diff
@@ -86,17 +83,20 @@ mod downstairs;
 mod upstairs;
 use upstairs::{UpCounters, UpstairsAction};
 
+mod io_limits;
+use io_limits::IOLimitGuard;
+
 /// Max number of write bytes between the upstairs and an offline downstairs
 ///
-/// If we exceed this value, the upstairs will give
-/// up and mark the offline downstairs as faulted.
-const IO_OUTSTANDING_MAX_BYTES: u64 = 1024 * 1024 * 1024; // 1 GiB
+/// If we exceed this value, the upstairs will give up and mark the offline
+/// downstairs as faulted.
+const IO_OUTSTANDING_MAX_BYTES: u64 = 50 * 1024 * 1024; // 50 MiB
```

Comment:
Where did this number come from?

Reply:
Mostly vibes!  This was previously the point at which backpressure delays started.

Reply:
To be clear, I don't think this is a bad number! Just wondering, esp when you were looking at the graphs showing the buffering cliff that real devices had.

Reply:
One data point: with 10x `crutest` instances doing 4K random writes on a single Gimlet (so 30x Downstairs), we end up buffering about 8 seconds of writes:
![Screenshot 2024-11-12 at 10 31 47 AM](https://github.com/user-attachments/assets/ff82fe9f-1056-4c2f-a1ac-89fa660a9131)

(there's 60 seconds of IO, then you can watch the number of active jobs drain)

This is a significant improvement from the ~50 seconds that we see on `main`:
![Screenshot 2024-11-12 at 10 31 57 AM](https://github.com/user-attachments/assets/b2a01f10-1eaa-404a-b350-cb5e8cd62991)

It may still be too much, but it's easy to tune these values later.